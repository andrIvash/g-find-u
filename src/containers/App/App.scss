@import '../../styles/Base.scss';

#app {
    display: flex;
    flex-direction: column;
    box-sizing: border-box;
    user-select: none;
    align-items: center;
    justify-content: center;

    &.has-border {
        border: var(--app-border-color);
    }
}