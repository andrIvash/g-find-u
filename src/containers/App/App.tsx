import React from 'react';
import { Routes, Route } from 'react-router-dom';
import HomePage from '../../components/HomePage';
import NoMatch from '../../components/NoMatch';

import './App.scss';

export const App = () => {
    return (
        <Routes>
            <Route path="/" element={<HomePage />} />
            <Route path="*" element={<NoMatch />} />
        </Routes>
    );
};

