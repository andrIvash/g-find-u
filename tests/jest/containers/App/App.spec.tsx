import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import '@testing-library/jest-dom';
import App from '../../../../src/containers/App';

jest.mock('../../../../src/containers/HomePage', () => () => <div>HomePage Mock</div>);
jest.mock('../../../../src/components/NoMatch', () => () => <div>NoMatch Mock</div>);

describe('App Component', () => {
    it('renders HomePage at root route', () => {
        render(
            <MemoryRouter initialEntries={['/']}>
                <App />
            </MemoryRouter>
        );
        expect(screen.getByText('HomePage Mock')).toBeInTheDocument();
    });


    it('renders NoMatch at an undefined route', () => {
        render(
            <MemoryRouter initialEntries={['/random']}>
                <App />
            </MemoryRouter>
        );
        expect(screen.getByText('NoMatch Mock')).toBeInTheDocument();
    });

    it('matches snapshot', () => {
        const { asFragment } = render(
            <MemoryRouter>
                <App />
            </MemoryRouter>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});