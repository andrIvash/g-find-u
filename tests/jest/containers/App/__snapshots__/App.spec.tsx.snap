// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="screen "
  >
    <div
      class="screen__controls center "
    >
      <div
        class="item"
      >
        <img
          class="icon rotate"
          src=""
        />
         React
      </div>
      <div
        class="simpleSwitch"
        data-testid="SimpleButton"
      >
        <div
          class="simpleSwitch__label simpleSwitch__leftLabel"
        >
          <img
            class="icon"
            src=""
          />
        </div>
        <span
          class="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
        >
          <span
            class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorDefault PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorDefault css-1i8lvs-MuiButtonBase-root-MuiSwitch-switchBase"
          >
            <input
              aria-label="controlled"
              class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
              type="checkbox"
            />
            <span
              class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
            />
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
          />
        </span>
        <div
          class="simpleSwitch__label simpleSwitch__rightLabel"
        >
          <img
            class="icon"
            src=""
          />
        </div>
      </div>
      <button>
        Dark Theme
      </button>
    </div>
    <div
      class="header "
    >
      <h2>
        Header
      </h2>
      <p>
        test
      </p>
      <div
        class="simpleButton"
        data-testid="SimpleButton"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Search
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <div
      class="screen__content"
    >
      <div
        data-testid="HomePage"
      >
        <h2>
          Home
        </h2>
      </div>
    </div>
    <div
      class="footer"
    >
      <h2>
        Footer
      </h2>
      <p>
        test
      </p>
    </div>
  </div>
</DocumentFragment>
`;
